src/SOS/lldbplugin/services.cpp

```diff
@@ -2575,9 +2592,15 @@ LLDBServices::GetVersionStringFromSection(lldb::SBTarget& target, lldb::SBSectio
         else if (sectionType == lldb::eSectionTypeData)
         {
             lldb::addr_t address = section.GetLoadAddress(target);
-            uint32_t size = section.GetByteSize();
-            if (SearchVersionString(address, size, versionBuffer, VersionBufferSize)) {
-                return true;
+            if (address != LLDB_INVALID_ADDRESS)
+            {
+#if !defined(__APPLE__)
+                address -= section.GetFileOffset();
```

Comment:
So this becomes a disk-layout address?

Reply:
I wouldn't call it disk-layout address, but it the same as above. It is more accurate. It works because on the older lldb's the file offset returned is pretty small so the resulting address was close enough to the being of the segment to find the version string signature.